src/app/main.rs

```diff
@@ -360,50 +365,61 @@ where
+                                            get_original_dst.clone(),
+                                            local_identity.clone(),
+                                            Admin::new(report, readiness, trace_level)
+                                                .into_accept(),
+                                        ),
+                                        drain_rx.clone(),
+                                    );
+                                },
                             );
 
                             if let Some((listener, tap_svc_name)) = control_listener {
-                                trace!("spawning tap server");
-                                task::spawn(tap_daemon.map_err(|_| ()));
-                                serve::spawn(
-                                    "tap",
-                                    listener,
-                                    tls::AcceptTls::new(
-                                        get_original_dst.clone(),
-                                        local_identity,
-                                        tap::AcceptPermittedClients::new(
-                                            std::iter::once(tap_svc_name),
-                                            tap_grpc,
-                                        ),
-                                    ),
-                                    drain_rx.clone(),
+                                info_span!("tap", local_addr=%listener.local_addr()).in_scope(
+                                    || {
+                                        trace!("spawning");
+                                        tokio::spawn(tap_daemon.map_err(|_| ()));
+                                        serve::spawn(
+                                            listener,
+                                            tls::AcceptTls::new(
+                                                get_original_dst.clone(),
+                                                local_identity,
+                                                tap::AcceptPermittedClients::new(
+                                                    std::iter::once(tap_svc_name),
+                                                    tap_grpc,
+                                                ),
+                                            ),
+                                            drain_rx.clone(),
+                                        );
+                                    },
                                 );
                             } else {
                                 drop((tap_daemon, tap_grpc));
                             }
 
-                            trace!("spawning dns resolver");
-                            task::spawn(logging::admin().bg("dns-resolver").future(dns_bg));
+                            info_span!("dns").in_scope(|| {
+                                trace!("spawning");
+                                tokio::spawn(dns_bg);
```

Comment:
Is the "dns" span actually propagated to `dns_bg` or does this need an `instrument`?